client/product_details/ProductDetails.jsx

```diff
@@ -11,9 +11,44 @@ import ProductOverview from './ProductOverview.jsx';
 class ProductDetails extends React.Component {
   constructor(props) {
     super(props);
+    
+    fetch(`http://localhost:8080/products/${this.props.product_id}`)
+      .then((response) => {
+        return response.json();
+      }) 
+      .then((data) => {
+        this.setState({
+          info: data,
+        });
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+    
+    fetch(`http://localhost:8080/products/${this.props.product_id}/styles`)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        for (var i = 0; i < data.results.length; i++) {
+          if (data.results[i]['default?']) {
+            var selectedStyle = data.results[i];
+          }
+        }
+        this.setState({
+          styleInfo: data.results,
+          selectedStyle: selectedStyle,
+        });
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+
     this.state = {
       view: 'default',
-      styleSelected: ''
+      selectedStyle: undefined,
+      info: undefined,
+      styleInfo: undefined,
     };
   }
   
```

Comment:
Def curious to talk about fetching from the constructor. I have it the same way as you do with .setState called in the .then() blocks. Leslie told me that wasn't how to do it and I'm confused!